ranger/config/commands.py

```diff
@@ -1959,3 +1959,46 @@ def tab(self, tabnum):
             in sorted(self.modes.keys())
             if mode
         )
+
+
+class confirm(Command):
+    """:confirm <command_to_be_confirmed> + <potential command arguments>
+
+        Can be used with any command. Ranger will ask if it should execute the
+        given command.
+    """
+
+    def execute(self):
+        from functools import partial
+
+        command = ''
+        if len(self.args) > 1 and self.arg(1) != 'chain':
+            for index in range(1, len(self.args)):
+                if index != len(self.args):
+                    command += self.args[index] + ' '
+            self.fm.ui.console.ask(
+                "Confirm execution of: %s (y/N)" % ''.join(command),
+                partial(self._question_callback, command),
+                ('n', 'N', 'y', 'Y'),
+            )
+        else:
+            arguments = self.rest(2).split(';')
```

Comment:
Let's use a list comprehension here instead.

Reply:
I already tried that, I did something like `arguments = [arg.strip(';') for arg in self.args]`. But this separates arguments of a command from the command itsself, e.g. If you do `:confirm chain delete; echo hi` you would geld a list with `['confirm', 'chain', 'delete', 'echo', 'hi']`. 
So what benefit would a list comprehension have?

Please don't get me wrong I will try my best to make this PR fit the needs of ranger, just curious about the reasons in order to learn something :)

Reply:
Okay figured it out. Just had to iterate over self.rest(2).split(';') in the comprehension. Sometimes I fail to see the obvious. 
:)